packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -234,11 +371,211 @@ class ShortcutMapProperty extends DiagnosticsProperty<Map<LogicalKeySet, Intent>
+           && !identical(trigger, LogicalKeyboardKey.controlRight)
+           && !identical(trigger, LogicalKeyboardKey.shift)
+           && !identical(trigger, LogicalKeyboardKey.shiftLeft)
+           && !identical(trigger, LogicalKeyboardKey.shiftRight)
+           && !identical(trigger, LogicalKeyboardKey.alt)
+           && !identical(trigger, LogicalKeyboardKey.altLeft)
+           && !identical(trigger, LogicalKeyboardKey.altRight)
+           && !identical(trigger, LogicalKeyboardKey.meta)
+           && !identical(trigger, LogicalKeyboardKey.metaLeft)
+           && !identical(trigger, LogicalKeyboardKey.metaRight));
+
+  /// The non-modifier key of the shortcut that is pressed after all modifiers
+  /// to activate the shortcut.
+  ///
+  /// For example, for `Control + C`, [trigger] should be
+  /// [LogicalKeyboardKey.keyC].
+  final LogicalKeyboardKey trigger;
+
+  /// Whether either (or both) control key should be held for [trigger] to
+  /// activate the shortcut.
+  ///
+  /// If false, then all control keys must be released at the event.
+  ///
+  /// See also:
+  ///
+  ///  * [LogicalKeyboardKey.controlLeft], [LogicalKeyboardKey.controlRight].
+  final bool control;
+
+  /// Whether either (or both) shift key should be held for [trigger] to
+  /// activate the shortcut.
+  ///
+  /// If false, then all shift keys must be released at the event.
+  ///
+  /// See also:
+  ///
+  ///  * [LogicalKeyboardKey.shiftLeft], [LogicalKeyboardKey.shiftRight].
+  final bool shift;
+
+  /// Whether either (or both) alt key should be held for [trigger] to
+  /// activate the shortcut.
+  ///
+  /// If false, then all alt keys must be released at the event.
+  ///
+  /// See also:
+  ///
+  ///  * [LogicalKeyboardKey.altLeft], [LogicalKeyboardKey.altRight].
+  final bool alt;
+
+  /// Whether either (or both) meta key should be held for [trigger] to
+  /// activate the shortcut.
+  ///
+  /// If false, then all meta keys must be released at the event.
+  ///
+  /// See also:
+  ///
+  ///  * [LogicalKeyboardKey.metaLeft], [LogicalKeyboardKey.metaRight].
+  final bool meta;
+
+  @override
+  Iterable<LogicalKeyboardKey>? get triggers sync* {
+    yield trigger;
+  }
+
+  @override
+  bool accepts(RawKeyEvent event, RawKeyboard state) {
+    final Set<LogicalKeyboardKey> pressed = state.keysPressed;
+    return event is RawKeyDownEvent
+      && (control == (pressed.contains(LogicalKeyboardKey.controlLeft) || pressed.contains(LogicalKeyboardKey.controlRight)))
+      && (shift == (pressed.contains(LogicalKeyboardKey.shiftLeft) || pressed.contains(LogicalKeyboardKey.shiftRight)))
+      && (alt == (pressed.contains(LogicalKeyboardKey.altLeft) || pressed.contains(LogicalKeyboardKey.altRight)))
+      && (meta == (pressed.contains(LogicalKeyboardKey.metaLeft) || pressed.contains(LogicalKeyboardKey.metaRight)));
+  }
+
+  /// Returns a description of the key set that is short and readable.
```

Comment:
Maybe use a different term than "key set", to avoid confusion with `LogicalKeySet`?

Reply:
You're right. I'm using "key combination".